File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/movimentacao/MovimentacaoFinanceiraGridDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.movimentacao;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class MovimentacaoFinanceiraGridDTO implements GridDTO {
    private UUID id;
    private String tituloDescricao;
    private String contaBancariaNome;
    private String tipo;
    private String formaPagamento;
    private BigDecimal valor;
    private LocalDate data;
    private java.util.UUID unidadeNegocioId;
    private String unidadeNegocioNome;
    private Boolean deleted;
}

File: src/frontend/src/app/components/financeiro/movimentacao-financeira/model/movimentacao-financeira-grid.dto.ts
export class MovimentacaoFinanceiraGridDTO {
  id: string;
  data: string;
  valor: number;
  tipo: string;
  unidadeNegocioId: string;
  unidadeNegocioNome: string;
  contaBancaria?: string;
  contaBancariaNome?: string;
  deleted?: boolean;

  constructor(
    id: string,
    data: string,
    valor: number,
    tipo: string,
    unidadeNegocioId: string,
    unidadeNegocioNome: string,
    contaBancaria?: string,
    contaBancariaNome?: string,
    deleted?: boolean
  ) {
    this.id = id;
    this.data = data;
    this.valor = valor;
    this.tipo = tipo;
    this.unidadeNegocioId = unidadeNegocioId;
    this.unidadeNegocioNome = unidadeNegocioNome;
    this.contaBancaria = contaBancaria;
    this.contaBancariaNome = contaBancariaNome;
    this.deleted = deleted;
  }
}
